pytket/pytket/backends/backendinfo.py

```diff
@@ -143,7 +143,11 @@ def from_dict(cls, d: Dict[str, Any]) -> "BackendInfo":
         :rtype: BackendInfo
         """
         args = dict(**d)
-        args["architecture"] = Architecture.from_dict(args["architecture"])
+        arch = args["architecture"]
+        if "links" in arch:
+            args["architecture"] = Architecture.from_dict(args["architecture"])
+        else:
+            args["architecture"] = FullyConnected.from_dict(args["architecture"])
```

Comment:
Could we store a name or ID of the Architecture class used, so that it is easily extensible?

Reply:
This would mean it would also work for RingArch and SquareArch.

Reply:
(this should probably be in serialization code, outside of `BackendInfo`)

Reply:
Yup, I think this would make sense if we wanted to introduce special handling for the other types, but maybe a bit too much complexity just to handle `FullyConnected`?